### PR TITLE
Add a new sample - dynamic line rasterization

### PR DESCRIPTION
**Pull request by Krzysztof-Dmitruk-Mobica**
## Description

This pull request proposes a sample that demonstrates methods for dynamically customizing the appearance of rendered lines.

### Notes
Tested on:

Ubuntu 23.04
Intel i7-11800H
Intel TigerLake-H GT1/Nvidia T600 Mobile
32 GB RAM

Please ensure the following points are checked:

- [X] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [X] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [X] I have commented any added functions (in line with Doxygen)
- [X] I have commented any code that could be hard to understand
- [X] My changes do not add any new compiler warnings
- [X] My changes do not add any new validation layer errors or warnings
- [X] I have used existing framework/helper functions where possible
- [X] My changes do not add any regressions
- [X] I have tested every sample to ensure everything runs correctly
- [X] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [X] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
  
## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [X] I have tested the sample on at least one compliant Vulkan implementation
- [X] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [X] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [X] Any dependent assets have been merged and published in downstream modules
- [X] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)
- [X] For new samples, I have added a tutorial README.md file to guide users through what they need to know to implement code using this feature. For example, see [conditional_rendering](./../samples/extensions/conditional_rendering)

### CODE REVIEW

**Review by SaschaWillems — CHANGES_REQUESTED** [review]
Thank you very much for this great sample! Code and sample look fine to me and work as expected on my setup (Win11 + RTX 2060).

I only have a minor request: Can you add a screenshot of the sample to the readme?

**Comment by Krzysztof-Dmitruk-Mobica** [comment]
Sure! I've added a screenshot.

File: samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+	cube_index_count                 = static_cast<uint32_t>(cube_indices.size());
+	edges_index_count                = static_cast<uint32_t>(edges_indices.size());
+	uint32_t vertex_buffer_size      = vertices.size() * sizeof(glm::vec3);
+	uint32_t cube_index_buffer_size  = cube_indices.size() * sizeof(uint32_t);
+	uint32_t edges_index_buffer_size = edges_indices.size() * sizeof(uint32_t);
+
+	vertex_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                    vertex_buffer_size,
+	                                                    VK_BUFFER_USAGE_VERTEX_BUFFER_BIT,
+	                                                    VMA_MEMORY_USAGE_GPU_TO_CPU);
```
Comment (asuessenbach) [inline samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp]: You probably mean `VMA_MEMORY_USAGE_CPU_TO_GPU` ?

File: samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+	vertex_buffer->update(vertices.data(), vertex_buffer_size);
+
+	cube_index_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                        cube_index_buffer_size,
+	                                                        VK_BUFFER_USAGE_INDEX_BUFFER_BIT,
+	                                                        VMA_MEMORY_USAGE_GPU_TO_CPU);
```
Comment (asuessenbach) [inline samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp]: As above.

File: samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+	cube_index_buffer->update(cube_indices.data(), cube_index_buffer_size);
+
+	edges_index_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                         edges_index_buffer_size,
+	                                                         VK_BUFFER_USAGE_INDEX_BUFFER_BIT,
+	                                                         VMA_MEMORY_USAGE_GPU_TO_CPU);
```
Comment (asuessenbach) [inline samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp]: As above

**Review by asuessenbach — COMMENTED** [review]
Besides one minor issue (three times), it's a great example!

File: samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp (hunk 1 above)
Comment (Krzysztof-Dmitruk-Mobica) [reply samples/extensions/dynamic_line_rasterization/dynamic_line_rasterization.cpp]: Fixed, thank you for noticing that.

**Review by asuessenbach — APPROVED** [review]

**Comment by gary-sweet** [comment]
Unfortunately I can't test this. My hardware doesn't support stippled lines or dynamic polygon mode.

It would be great if the sample could check the support for each feature and just not use any that weren't supported rather than requiring everything in order to run anything.

**Review by SaschaWillems — APPROVED** [review]

**Pull request merged**